src/d3a/models/area/__init__.py

```diff
@@ -388,13 +390,14 @@ def _consume_commands_from_aggregator(self):
     def tick(self):
         self._consume_commands_from_aggregator()
 
-        if ConstSettings.IAASettings.MARKET_TYPE == 2 or \
-                ConstSettings.IAASettings.MARKET_TYPE == 3:
+        if ConstSettings.IAASettings.MARKET_TYPE == 2:
             if ConstSettings.GeneralSettings.EVENT_DISPATCHING_VIA_REDIS:
                 self.dispatcher.publish_market_clearing()
             else:
-                for market in self.all_markets:
-                    market.match_offers_bids()
+                if self.next_market is not None:
+                    self.matcher.calculate_recommendation(
+                        *self.next_market.get_open_bids_and_offers)
+                    self.next_market.match_recommendation(self.matcher.bid_offer_pairs)
```

Comment:
Why not repeating the same process for all future markets ? Essentially you can do the following and avoid the if:

```suggestion
                for market in self.all_markets:
                    self.matcher.calculate_recommendation(
                        *market.get_open_bids_and_offers)
                    market.match_recommendation(self.matcher.bid_offer_pairs)
```